Modules/_testcapimodule.c

```diff
@@ -5076,6 +5187,10 @@ static PyMethodDef TestMethods[] = {
     {"get_delta_fromdsu",        get_delta_fromdsu,              METH_VARARGS},
     {"get_date_fromtimestamp",   get_date_fromtimestamp,         METH_VARARGS},
     {"get_datetime_fromtimestamp", get_datetime_fromtimestamp,   METH_VARARGS},
+    {"get_fields_from_date",       get_fields_from_date,         METH_VARARGS},
```

Comment:
I suggest to rename these functions to have a name closer to their macro. Example:

```suggestion
    {"PyDateTime_GET",       test_PyDateTime_GET,         METH_VARARGS},
```

* get_fields_from_date => PyDateTime_GET
* get_fields_from_datetime => PyDateTime_DATE_GET
* get_fields_from_time => PyDateTime_TIME_GET
* get_fields_from_delta => PyDateTime_DELTA_GET

Currently, I'm very confused by "from_date" which uses "PyDateTime_GET", whereas "from_datetime" uses "PyDateTime_DATE_GET".

Note: I know that there are existing "get_xxx" helper functions. I'm fine with leaving existing names unchanged. But I would prefer to not add futher confusing names. _testcapi is used by the whole stdlib, I prefer to use reuse names of the C API.

Reply:
I'm not sure I agree with this change. These are Python wrapper functions for the C API, and they don't do the same thing as anything in the C API - they construct a tuple of Python objects rather than retrieving the fields from the C structs. As such, I think it makes sense to use a Python naming convention.

I will concede, though, that it does make it clearer what these are intended to test just from reading the names.

> _testcapi is used by the whole stdlib, I prefer to use reuse names of the C API.

Is this true? Are we only supposed to be putting test support functions in here? If so, I think we may want to move the datetime C API test functions into a dedicated module, because I've been using this file as explicit wrappers for *specific tests*.

Reply:
Sorry, I wanted to write _testcapi is used to **test** the whole stdlib.

I prefer names starting with "PyDateTime_" to show that these functions are related to the datetime module. The relationship between "get_fields_from_delta" and "datetime" in less obvious.

> If so, I think we may want to move the datetime C API test functions into a dedicated module, because I've been using this file as explicit wrappers for specific tests.

_testcapi became really large: 6154 lines of C code. It tests a lot of unrelated things. I'm open to discuss creating a new separated C extension just to test the datetime module, but that should be done in a separated issue and separated PR.